### ISSUE TITLE
Population can exceed the configured populationSize by 3× when heavy-pool results land in one generation

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#3508]
## Observed

CI shard 6 of the Test Coverage run on PR #3503 reported a fitness queue depth
of **48** for a run configured with `populationSize: 15`
(`test/config/ThroughputMetrics.ts`, generation 1 of a 3-iteration evolve).
Locally the same run peaks at ~18.

## Analysis

`src/NEAT/NeatEvolution.ts` assembles the next population by concatenation:

```ts
neat.population = [
  ...elitists,
  ...trainedPopulation,
  ...fineTunedPopulation,
  ...newPopulation,
  ...dnaPopulation,
];
```

Only the bred slice is budgeted — `newPopSize = effectivePopSize - elitists -
dna - trainingComplete - discoveryComplete - fineTuned - 1 - newPopulation` — and
it simply floors at zero (`breedBatch` returns `[]` for `count <= 0`). The other
slices are not capped, so when several heavy-pool training / fine-tuning /
discovery tasks complete in the same generation (much more likely on a slow or
contended runner) the population, and therefore the fitness queue, grows well
beyond the configured size.

## Impact

- Memory and CPU per generation scale above what the operator configured.
- Throughput telemetry (`fastQueueMaxDepth`, `scoredCreatureCount`) reports
  depths that look anomalous but are "correct" for the state.
- It made a coverage-gate assertion flaky; PR #3503 relaxed that assertion to
  bound the metric by work actually done rather than by `2 × populationSize`.

## What a fix looks like

Decide whether exceeding `effectivePopSize` is intended. If not, trim the
concatenated population back to `effectivePopSize` (dropping the weakest
non-elite creatures) after assembly, and add a test asserting
`population.length <= effectivePopulationSize` across a multi-generation run
with training/fine-tuning active. If it *is* intended, document the bound
(e.g. worst-case multiple of `populationSize`) so telemetry consumers and tests
can assert against it.

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/3528 has been created to address this issue.

**Comment by VibeCoderST** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/3528 has been created to address this issue.

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/3529 has been created to address this issue.
